src/SearchParameters/index.js

```diff
@@ -459,36 +462,33 @@ SearchParameters._parseNumbers = function(partialState) {
     'minProximity'
   ];
 
-  forEach(numberKeys, function(k) {
-    var value = partialState[k];
-    if (isString(value)) numbers[k] = parseFloat(partialState[k]);
+  // parse numeric root keys to floats
+  newState = mapValues(newState, function(value, key) {
```

Comment:
Actually I tried, and I tried your suggestion of trying to return early without cloning if we detect that there is nothing to do, and it was even slower :/